pallets/subspace/src/subnet.rs

```diff
@@ -362,6 +362,10 @@ impl<T: Config> Pallet<T> {
         SubnetNames::<T>::iter().find(|(_, n)| n == name).map(|(id, _)| id)
     }
 
+    pub fn get_subnet_name(netuid: u16) -> Vec<u8> {
+        Self::subnet_params(netuid).name
+    }
```

Comment:
This function is reaaaaally inefficient. If someone wishes to fetch only the name of a subnet, let them access it directly:

```suggestion
    pub fn get_subnet_name(netuid: u16) -> Vec<u8> {
        SubnetNames::<T>::get(netuid)
    }
```

I would be even better if we didn't need this function at all. I'd say we access SubnetNames directly when we need to find a name.